test/torchaudio_unittest/functional/functional_cpu_test.py

```diff
@@ -70,47 +70,104 @@ def test_pitch(self, frequency):
         self.assertFalse(s)
 
 
-class TestDB_to_amplitude(common_utils.TorchaudioTestCase):
-    def test_DB_to_amplitude(self):
-        # Make some noise
-        x = torch.rand(1000)
+class Testamplitude_to_DB(common_utils.TorchaudioTestCase):
+    AMPLITUDE_MULT = 20.
+    POWER_MULT = 10.
+    AMIN = 1e-10
+    REF = 1.0
+    # FIXME: So the DB multiplier is 0?
+    DB_MULT = math.log10(max(AMIN, REF))
+
+    def _make_spectrogram(self, items, channels, wave_samples, scale=1.):
+        # Make some noise, -1 to 1 (then scale it)
+        wave = (torch.rand(items, channels, wave_samples) * 2. - 1.) * scale
         spectrogram = torchaudio.transforms.Spectrogram()
-        spec = spectrogram(x)
-
-        amin = 1e-10
-        ref = 1.0
-        db_multiplier = math.log10(max(amin, ref))
-
-        # Waveform amplitude -> DB -> amplitude
-        multiplier = 20.
-        power = 0.5
-
-        db = F.amplitude_to_DB(torch.abs(x), multiplier, amin, db_multiplier, top_db=None)
-        x2 = F.DB_to_amplitude(db, ref, power)
-
-        torch.testing.assert_allclose(x2, torch.abs(x), atol=5e-5, rtol=1e-5)
+        return spectrogram(wave)
 
+    def _ensure_reversible(self, spec):
+        """Check `amplitude_to_db` returns the original when reversed."""
         # Spectrogram amplitude -> DB -> amplitude
-        db = F.amplitude_to_DB(spec, multiplier, amin, db_multiplier, top_db=None)
-        x2 = F.DB_to_amplitude(db, ref, power)
+        db = F.amplitude_to_DB(spec, self.AMPLITUDE_MULT, self.AMIN, self.DB_MULT, top_db=None)
+        x2 = F.DB_to_amplitude(db, self.REF, 0.5)
 
         torch.testing.assert_allclose(x2, spec, atol=5e-5, rtol=1e-5)
```

Comment:
Please replace `torch.testing.assert_allclose` with `self.assertEqual`. The use of `assert_allclose` is discouraged now.